test/0.8.9/self-owned-steth-burner.test.js

```diff
@@ -804,6 +804,33 @@ contract('SelfOwnedStETHBurner', ([appManager, voting, deployer, depositor, anot
       assertRevert(burner.recoverERC20(mockERC20Token.address, bn(1), { from: deployer }), `ERC20: transfer amount exceeds balance`)
     })
 
+    it(`can't recover stETH via ERC721(NFT)`, async () => {
+      // initial stETH balance is zero
+      assertBn(await lido.balanceOf(anotherAccount), stETH(0))
+      // submit 10 ETH to mint 10 stETH
+      await web3.eth.sendTransaction({ from: anotherAccount, to: lido.address, value: ETH(10) })
+      // check 10 stETH minted on balance
+      assertBn(await lido.balanceOf(anotherAccount), stETH(10))
+      // transfer 5 stETH to the burner account
+      await lido.transfer(burner.address, stETH(5), { from: anotherAccount })
+      // transfer 5 stETH to voting
+      await lido.transfer(voting, stETH(5), { from: anotherAccount })
+
+      // request 5 stETH to be burned later
+      await lido.approve(burner.address, stETH(5), { from: voting })
+      await burner.requestBurnMyStETH(stETH(5), { from: voting })
+
+      // check balances one last time
+      assertBn(await lido.balanceOf(anotherAccount), stETH(0))
+      assertBn(await lido.balanceOf(voting), stETH(0))
+      assertBn(await lido.balanceOf(burner.address), stETH(10))
```

Comment:
It's not clear for me, why we need to put 10 stETH and trying to recover only 1 and why a such complicated way was used to top up the burner.

Reply:
Hey, addressed the issues within the updated code, thank you :pray: 

The top up process is rather complicated because of design requirements (only voting can request the burning & explicit allowance required).

Please check out the update code :+1: